app/controllers/ams_subjects_controller.rb

```diff
@@ -0,0 +1,38 @@
+class AmsSubjectsController < ApplicationController
+  before_action :set_subject_category
+  before_action :set_ams_subject, only: %i[edit update destroy]
+
+  def edit; end
+
+  def update
+    if @ams_subject.update(ams_subject_params)
+      redirect_to subject_category_url(@subject_category),
+                  notice: 'Ams Subject was successfully updated'
```

Comment:
AMS is an acronym for American Mathematical Society. Please use "AMS" in flash messages, instead of Ams or ams.